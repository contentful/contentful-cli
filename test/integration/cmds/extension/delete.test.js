const nixt = require('nixt')
const { resolve } = require('path')

const bin = resolve(__dirname, './../../../../', 'bin')

const app = () => {
  return nixt({ newlines: true })
    .cwd(bin)
    .base('./contentful.js ')
    .clone()
}

test('should print help message', done => {
  app()
    .run('extension delete --help')
    .code(0)
    .expect(result => {
      const resultText = result.stdout.trim()
      expect(resultText).toMatchSnapshot('help data is incorrect')
    })
    .end(done)
})

test('should exit 1 when no args given except space id', done => {
  app()
    .run('extension delete')
    .code(1)
    .expect(result => {
      const regex = /Missing required argument:\s+id/
      expect(result.stderr.trim()).toMatch(regex)
    })
    .end(done)
})
