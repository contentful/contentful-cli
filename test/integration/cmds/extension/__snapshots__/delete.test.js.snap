// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should print help message: help data is incorrect 1`] = `
"extension delete

Delete an extension

Options:
  -h, --help                Show help                                  [boolean]
  --id                      Extension id                     [string] [required]
  --management-token, --mt  Contentful management API token             [string]
  --space-id                Space id                                    [string]
  --environment-id          Environment id                              [string]
  --version                 Current version of the extension for optimistic
                            locking                                     [number]
  --force                   Force operation without explicit version   [boolean]

Copyright 2019 Contentful"
`;
